### ISSUE TITLE
[BUG][torch-spyre]: `torch.exp` miscomputes on fp16 tensors whose row width is not a multiple of `elem_in_stick`

### ISSUE THREAD

**Issue by abhishekkunuru6-cmyk** [torch-spyre/torch-spyre#3799]
### What happened?

`torch.exp` on a plain, unsliced fp16 tensor whose last dim is not a multiple of
64 returns wrong results, or fails in the backend. No view or `storage_offset`
involved.

At `SENCORES=32`, `(63,100)` and `(64,100)` are wrong by 45.88 and 29.22
(tolerance `atol=0.1, rtol=0.1`); `(3,100)` and `(4,100)` instead fail to
compile:

```
sbf-ddc: DtException: Cannot currently apply coordinate masking for dimensions
split across cores, file .../deeptools/ddc/ddcv1.cpp line 3516
```

At `SENCORES=1`, `(3,100)` is wrong by 10.34.

### What did you expect to happen?

`torch.compile(torch.exp)(x)` matches CPU for any shape.

### How can we reproduce it?

```python
import torch
import torch_spyre  # noqa: F401

torch.manual_seed(0)

def check(shape):
    b = torch.randn(*shape, dtype=torch.float16)
    ref = torch.exp(b.clone()).float()
    torch._dynamo.reset()
    try:
        got = torch.compile(torch.exp, backend="inductor", dynamic=False)(
            b.clone().to("spyre")
        ).cpu().float()
    except Exception as e:   # some shapes fail to compile; keep going
        print(f"{str(shape):10s}: RAISED {type(e).__name__}: {str(e)[:70]}")
        return
    d = (got - ref).abs().max().item()
    ok = torch.allclose(got, ref, atol=0.1, rtol=0.1)
    print(f"{str(shape):10s}: {'PASS' if ok else 'FAIL'}  max_abs_diff={d:.4f}")

for shape in [(3, 100), (4, 100), (63, 100), (64, 100),   # padded rows
              (5, 128), (64, 128)]:                       # controls
    check(shape)
```

`TORCHINDUCTOR_FORCE_DISABLE_CACHES=1`, at both `SENCORES=1` and `SENCORES=32`.

### Any environmental details we need to know?

Output of the script above (`torch.manual_seed(0)`, so these are reproducible):

|                | SENCORES=1  | SENCORES=32 |
|---|---|---|
| `exp (3,100)`  | FAIL 30.27  | RAISED |
| `exp (4,100)`  | PASS 0.125  | RAISED |
| `exp (63,100)` | PASS 0.047  | FAIL 22.53 |
| `exp (64,100)` | PASS 0.078  | FAIL 29.05 |
| `exp (5,128)`  | PASS 0.031  | PASS 0.031 |
| `exp (64,128)` | PASS 0.078  | PASS 0.078 |

Padding is the only variable — `(64,128)` differs from `(64,100)` only in row
width and passes everywhere. Failing shapes are one to two orders of magnitude
outside the `atol=0.1, rtol=0.1` tolerance, so there is no ambiguity with fp16
rounding.

`x + x` is unaffected on the same shapes (max abs diff 0.0039). It hits
`can_use_same_layout` and reuses the input STL verbatim; `exp` goes through
`_single_arg_op_layout`, which re-selects the output layout. So the defect is in
the coordinate→address stage or work division, not coordinate derivation.
`superdsc.py`'s `dim_device_stride` is a plausible starting point.

Layouts, for reference — `device_size` is padded, `stride_map` is not:

```
(4, 100)  -> device_size=[2, 4, 64],  stride_map=[64, 100, 1]
(64, 100) -> device_size=[2, 64, 64], stride_map=[64, 100, 1]
(5, 128)  -> device_size=[2, 5, 64],  stride_map=[64, 128, 1]
```

Found while testing sliced placeholders, but not caused by `storage_offset`:
sliced results are exactly predicted by unsliced results of the same logical
shape in both core counts (`(4,100)[1:,:]` behaves as `(3,100)`,
`(64,100)[1:,:]` as `(63,100)`).

### Relevant log output

```Shell

```

**Comment by abhishekkunuru6-cmyk** [comment]
cc: @ani300 @saurabh-srivastava4 @goutham-binnadi-gopala

**Comment by yoheiueda** [comment]
This is a side effect of #3248. That PR fixed #3240, where a matmul accumulated the padding lanes of the last stick of its operand's stick dimension and `exp` over those lanes produced `0x7FFF`, the hardware NaN, which then made the matmul output wrong. That word is the one value the padding scheme does not tolerate: the other operand's padding is filled with zeros, so an ordinary large finite lane contributes nothing to the accumulation, while `0x7FFF` survives being multiplied by zero and reaches the sum. The fix extended coordinate masking (`SAMV`) from reductions to an allowlisted pointwise op, `exp`, seeding its padded output lanes with `-1e4` so `exp` underflows them to zero and the downstream contraction sees nothing.

Coordinate masking was built for reductions, and reductions have two properties this case does not. A reduction does not divide its stick dimension across cores, and its output holds a single stick on that dimension. `exp` is pointwise, so neither holds: the stick dimension is available for work division, and the output has as many sticks as the shape needs.

The first shows up as the compile reject, `Cannot currently apply coordinate masking for dimensions split across cores, ddcv1.cpp line 3519`, which fires when `numWkSlicesPerDim_['out'] > 1`. That one is loud. The second is silent, and no core setting avoids it.

Two settings are worth keeping apart. **Whether work division splits the stick dimension**, which is what the reject is about, and **whether core division is disabled altogether with `SENCORES=1`**. Neither avoids the wrong answers. Extents below hold the last stick at 16 real elements and 48 padding lanes, so stick count is the only shape variable, and `damaged` is the index of each stick that comes back wrong:

| rows | extent | sticks | `SENCORES=1` | damaged | `SENCORES=32` | damaged |
|---|---|---|---|---|---|---|
| 3 | 16 | **1** | **ok** | | **ok** | |
| 3 | 80 | 2 | 144 wrong | 0 | reject | |
| 3 | 144 | **3** | **ok** | | reject | |
| 3 | 208 | 4 | 432 wrong | 0, 1, 2 | reject | |
| 3 | 272 | **5** | **ok** | | reject | |
| 3 | 336 | 6 | 144 wrong | 4 | reject | |
| 3 | 400 | **7** | **ok** | | reject | |
| 3 | 464 | 8 | 432 wrong | 4, 5, 6 | reject | |
| 3 | 528 | **9** | **ok** | | reject | |
| 64 | 16 | **1** | **ok** | | **ok** | |
| 64 | 80 | 2 | **ok** | | 3072 wrong | 0 |
| 64 | 144 | **3** | **ok** | | **ok** | |
| 64 | 208 | 4 | **ok** | | 3072 wrong | 2 |
| 64 | 272 | **5** | **ok** | | **ok** | |
| 64 | 336 | 6 | **ok** | | 3072 wrong | 4 |
| 64 | 400 | **7** | **ok** | | **ok** | |
| 64 | 464 | 8 | **ok** | | 3072 wrong | 6 |
| 64 | 528 | **9** | **ok** | | **ok** | |

Even stick count is wrong and odd is correct, at both core counts, and a single stick on the stick dimension, which is the reduction shape, is always correct. Aligned extents (64, 128, 192, 256) are correct at every stick count and every core count. The damage size is `rows` times 48 padding lanes times the number of damaged sticks.

A damaged stick is an exact copy of the mask pattern that belongs to the last stick. Every one of them has lanes 0..15 correct and lanes 16..63 holding a single value, `5.96e-08`, which is `exp(-1e4)` at the device underflow floor, and `-1e4` is the mask value the pass seeds padding with. There are no partially damaged sticks and no other wrong values. Since extent 208 puts 16 real elements and 48 padding lanes in the last stick, the interval being applied is the correct one, `16..63`, and only the stick it is applied to is wrong. Row 0 of `(3,464)`, 8 sticks, with `.` correct and `M` the mask floor:

```
stick 0: ................................................................
stick 1: ................................................................
stick 2: ................................................................
stick 3: ................................................................
stick 4: ................MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
stick 5: ................MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
stick 6: ................MMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMMM
stick 7: ................------------ padding, not readable ---------------
```

Stick 7 is the one that should be masked, and its 16 real lanes come back correct. So the mask is applied to the wrong sticks instead of the right one, in a contiguous run ending one stick short of the target. At `SENCORES=32` with 64 rows that run is a single stick, `sticks - 2`. I cannot read the last stick's padding lanes from the host, so whether it is masked as well as the others is not something this measurement settles.

The SDSC agrees. At a logical extent of 65 it carries `coordinateMasking_: {'out': [[65, 63]]}`, which names the padding exactly, and the lanes that come back dead are 1..63.

The row count also matters in a way I cannot explain. `rows=64` is correct at one core and wrong at 32; `rows=3` is the reverse. At extent 80 with 32 cores, `(32,80)` is wrong by 1536, `(16,80)` is rejected, and `(128,80)` and `(256,80)` are correct.

So there are two questions for the back end. Whether applying the mask to the wrong sticks is a bug that can be fixed, since the SDSC it is given is correct and the failure appears with core division fully disabled. And whether coordinate masking can support a stick dimension that work division may split, which is what the reject asks for. If both can be addressed, this is a matter of extending the existing mechanism. If not, masking is not the route for pointwise ops and the padded lanes have to be handled where they are created, which is the direction #3290 points.
